service/frontend/redirection_interceptor.go

```diff
@@ -45,75 +45,75 @@ import (
 var (
 	dcRedirectionMetricsPrefix = "DCRedirection"
 
-	localAPIResults = map[string]respAllocFn{
+	localAPIResponses = map[string]responseConstructorFn{
```

Comment:
now that I'm looking at this, have we benchmarked this against the equivalent reflection-based code? I see we need the maps anyway but they could just be sets and use reflection to construct the response. I'm curious about code bloat effects

Reply:
I also thought about it. I even have a func for it:
```go
func newOfSameType(val reflect.Value) reflect.Value {
	valType := val.Type().Elem()
	newValue := reflect.New(valType)
	val.Set(newValue)
	return newValue
}
```
but as soon as map is needed anyway, I am in doubt. Best thing, I guess, is code-gen the map.

Reply:
not quite that easy, you have to go from grpc method name to first return value of that method on WorkflowService. but only like 3-4 more lines.

we can't just code-gen, the maps hold the semantics about which calls need to be forwarded. but they could be map[string]struct{}, and then use reflection to make the response values